azure-kusto-ingest/azure/kusto/ingest/_ingest_client.py

```diff
@@ -46,40 +50,39 @@ def ingest_from_dataframe(self, df, ingestion_properties):
 
         df.to_csv(temp_file_path, index=False, encoding="utf-8", header=False, compression="gzip")
 
-        fd = FileDescriptor(temp_file_path)
-
-        blob_name = "{db}__{table}__{guid}__{file}".format(
-            db=ingestion_properties.database, table=ingestion_properties.table, guid=uuid.uuid4(), file=file_name
-        )
+        ingestion_properties.format = DataFormat.CSV
 
-        containers = self._resource_manager.get_containers()
-        container_details = random.choice(containers)
-        storage_client = CloudStorageAccount(container_details.storage_account_name, sas_token=container_details.sas)
-        blob_service = storage_client.create_block_blob_service()
+        self.ingest_from_file(temp_file_path, ingestion_properties)
 
-        blob_service.create_blob_from_path(container_name=container_details.object_name, blob_name=blob_name, file_path=temp_file_path)
-
-        url = blob_service.make_blob_url(container_details.object_name, blob_name, sas_token=container_details.sas)
-
-        self.ingest_from_blob(BlobDescriptor(url, fd.size), ingestion_properties=ingestion_properties)
-
-        fd.delete_files()
         os.unlink(temp_file_path)
 
     def ingest_from_file(self, file_descriptor, ingestion_properties):
         """Enqueuing an ingest command from local files.
         :param file_descriptor: a FileDescriptor to be ingested.
         :param azure.kusto.ingest.IngestionProperties ingestion_properties: Ingestion properties.
         """
-        file_descriptors = list()
         containers = self._resource_manager.get_containers()
 
         if isinstance(file_descriptor, FileDescriptor):
             descriptor = file_descriptor
         else:
             descriptor = FileDescriptor(file_descriptor)
 
-        file_descriptors.append(descriptor)
+        if (
+            ingestion_properties.format == DataFormat.PARQUET
+            or ingestion_properties.format == DataFormat.ORC
+            or ingestion_properties.format == DataFormat.AVRO
+            or descriptor.path.endswith(".gz")
+            or descriptor.path.endswith(".zip")
+        ):
+            zipped_stream = open(descriptor.path, "rb")
+        else:
+            descriptor.stream_name += ".gz"
```

Comment:
I would extract this entire method..it has a very specific job, to read data from the descriptor (might as well be on the descriptor)